weaviate/collection/classes.py

```diff
@@ -0,0 +1,519 @@
+        if len(self.reference_class_name) > 1:
+            ref_collection_name += self.reference_class_name[1:]
+        ret_dict["dataType"] = [ref_collection_name]
+        return ret_dict
+
+
+class CollectionConfig(CollectionConfigBase):
+    properties: Optional[List[Union[Property, ReferenceProperty]]] = None
+
+    def model_post_init(self, __context: Any) -> None:
+        collection_name = self.name[0].upper()
+        if len(self.name) > 1:
+            collection_name += self.name[1:]
+        self.name = collection_name
+
+    def to_dict(self) -> Dict[str, Any]:
+        ret_dict = super().to_dict()
+
+        if self.properties is not None:
+            ret_dict["properties"] = [prop.to_dict() for prop in self.properties]
+
+        return ret_dict
+
+
+class MetadataGet(BaseModel):
+    vector: bool = False
+    distance: bool = False
+    certainty: bool = False
+    score: bool = False
+    explain_score: bool = Field(alias="explainScore", default=False)
+    is_consistent: bool = Field(alias="isConsistent", default=False)
+
+    def _get_fields(self) -> Set[str]:
+        additional_props: Set[str] = set()
+        for field, value in self.model_fields.items():
+            enabled: bool = getattr(self, field)
+            if enabled:
+                name = value.alias if value.alias is not None else field
+                additional_props.add(name)
+        return additional_props
+
+    def to_graphql(self) -> str:
+        additional_props = self._get_fields()
+        if len(additional_props) > 0:
+            return "_additional{" + " ".join(additional_props) + "}"
+        else:
+            return ""
+
+    def to_rest(self) -> str:
+        return ",".join(self._get_fields())
+
+
+@dataclass
+class MetadataReturn:
+    # uuid: Optional[uuid_package.UUID] = Field(None, alias="id")
+    # vector: Optional[List[float]] = None
+    # creation_time_unix: Optional[int] = Field(None, alias="creationTimeUnix")
+    # last_update_time_unix: Optional[int] = Field(None, alias="lastUpdateTimeUnix")
+    # distance: Optional[float] = None
+    # certainty: Optional[float] = None
+    # score: Optional[float] = None
+    # explain_score: Optional[str] = Field(None, alias="explainScore")
+    # is_consistent: Optional[bool] = Field(None, alias="isConsistent")
+    uuid: Optional[uuid_package.UUID] = None
+    vector: Optional[List[float]] = None
+    creation_time_unix: Optional[int] = None
+    last_update_time_unix: Optional[int] = None
+    distance: Optional[float] = None
+    certainty: Optional[float] = None
+    score: Optional[float] = None
+    explain_score: Optional[str] = None
+    is_consistent: Optional[bool] = None
+
+    def __init__(self, **data: Dict[str, Any]) -> None:
+        def _to_uuid(uuid_str: str) -> uuid_package.UUID:
+            return uuid_package.UUID(hex=uuid_str)
+
+        def _parse(key: str) -> Any:
+            value = data.get(key)
```

Comment:
do we know that the keys are always present?

Reply:
Not sure, are all the keys passed back from gRPC even if the query only specifies a subset of them (with the unspecified ones as `None`)?